File: server/src/main/java/org/dan/jadalnia/app/festival/order/pojo/OrderMem.java
package org.dan.jadalnia.app.festival.order.pojo;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.experimental.FieldDefaults;
import org.dan.jadalnia.app.user.Uid;

@Getter
@Builder
@FieldDefaults(makeFinal = true)
@RequiredArgsConstructor
public class OrderMem {
    Uid customerId;
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/order/pojo/OrderMem.kt
package org.dan.jadalnia.app.order.pojo

import org.dan.jadalnia.app.festival.menu.DishName
import org.dan.jadalnia.app.token.TokenPoints
import org.dan.jadalnia.app.user.Uid
import org.dan.jadalnia.util.collection.MapQ.QueueInsertIdx
import java.util.concurrent.atomic.AtomicReference

data class OrderMem(
    val customer: Uid,
    val state: AtomicReference<OrderState>,
    val label: OrderLabel,
    val insertQueueIdx: AtomicReference<QueueInsertIdx>,
    val cost: AtomicReference<TokenPoints>,
    val items: AtomicReference<List<OrderItem>>) {

  fun toContentMap(): Map<DishName, Int>
      = items.get().associate { item -> Pair(item.name, item.quantity) }
}
